pallets/settlement/src/lib.rs

```diff
@@ -215,7 +215,9 @@ pub struct Instruction<Moment, BlockNumber> {
     /// Date at which this instruction was created
     pub created_at: Option<Moment>,
     /// Date from which this instruction is valid
-    pub valid_from: Option<Moment>,
+    pub trade_date: Option<Moment>,
```

Comment:
isn't it something compulsory data needed for the instruction creation?  Same for the `value_date`.

We don't force the settle on the value_date or after that, But this data should be compulsory for the settlement instruction.

Reply:
I feel like, whilst we aren't really using them for on-chain computation, it is fine for them to be optional.

We will revisit them in the next major upgrade (along with logic to use them) so could reconsider then. Currently the STO module uses None (along with lots of test cases), so I'd be more inclined to leave the change as small rather than refactoring those.